### PR TITLE
feat(rust)!: Schema improvements

### PR DESCRIPTION
**Pull request by rben01**
Added docs to `Schema` methods
Added tests for `Schema` methods
Removed the `try_` methods from Schema because they were almost universally unused and didn't add anything that a user couldn't do for themself with `.ok_or_else()`
Made a bunch of quality of life changes, first and foremost implementing `FromIterator<impl Into<Field>> for Schema`, which gets us the ability to `collect()` an iterator of `PolarsResult<impl Into<Field>>` into a `PolarsResult<Schema>`. The big change that needed to occur for this to be possible was to change `Field: From<&ArrowField>` to `Field: From<ArrowField>`.
Also made some minor improvements to a handful `Schema` method implementations

### CODE REVIEW

**Comment by ritchie46** [comment]
> Removed the try_ methods from Schema because they were almost universally unused and didn't add anything that a user couldn't do for themself with 

I'd like to keep these. These will be used more in optimizer and this will create a uniform error message for them. And it is more elegant IMO to use a `try` if you know it might fail.

**Comment by rben01** [comment]
Is there a way to force a rerun of the checks without pushing an empty commit or closing and reopening the PR?

**Comment by ritchie46** [comment]
> Is there a way to force a rerun of the checks without pushing an empty commit or closing and reopening the PR?

Do a rebase and force push. :+1:

**Comment by rben01** [comment]
> > Is there a way to force a rerun of the checks without pushing an empty commit or closing and reopening the PR?
> 
> Do a rebase and force push. 👍

Closing and reopening the PR also works 😉

File: polars/polars-core/src/datatypes/field.rs [hunk 1]
```diff
@@ -137,8 +136,7 @@ impl From<&ArrowDataType> for DataType {
             ArrowDataType::Dictionary(_, _, _) => DataType::Categorical(None),
             #[cfg(feature = "dtype-struct")]
             ArrowDataType::Struct(fields) => {
-                let fields: Vec<Field> = fields.iter().map(|fld| fld.into()).collect();
-                DataType::Struct(fields)
+                DataType::Struct(fields.iter().cloned().map(|fld| fld.into()).collect())
```
Comment (ritchie46) [inline polars/polars-core/src/datatypes/field.rs:139]: Then we also don't have to clone before we convert.

**Comment by ritchie46** [comment]
Looks good. :+1:

**Pull request merged**